### PR TITLE
WT-9302 Improve parsing of unknown configuration fields

### PR DESCRIPTION
**Pull request by ruby-oujo**
* Created a list `deleted_config_options` that stores removed configuration options still being used in validity checks. 
  * The config string is checked against these items, if the item is present then we do not fail and instead warn. 
  * If the unknown config is not part of the list, fail as usual.

### CODE REVIEW

**Comment by sueloverso** [comment]
@ruby-oujo my initial thoughts would be that this would be part of `dist/api_data.py` and specific to the `file_meta` area. The source files in the `config` directory do not contain any knowledge of the actual configuration settings other than the auto-generated files. Anyone making changes to the configuration is modifying `dist/api_data.py`.

**Review by etienneptl — APPROVED** [review]
LGTM!

File: dist/api_data.py [hunk 1]
```diff
@@ -945,9 +948,9 @@ def __ge__(self, other):
         time in seconds to retain data on tiered storage on the local tier for faster read
         access''',
         min='0', max='10000'),
-    Config('object_target_size', '10M', r'''
-        the approximate size of objects before creating them on the tiered storage tier''',
-        min='100K', max='10TB'),
+    Config('object_target_size', '0', r'''
+        This option is no longer supported, retained for backward compatibility''',
+        min='0', undoc=True),
```
Comment (sueloverso) [inline dist/api_data.py:953]: I think we do want to remove this from the wiredtiger_open settings. The retained one is the one for the metadata.

**Review by sueloverso — APPROVED** [review]
LGTM but I think we do want to remove it from the connection config. The one we're retaining is in the file metadata.

**Comment by sueloverso** [comment]
@ruby-oujo  The failure in the PR testing is compatibility. That should be investigated before merging.

File: dist/api_data.py (hunk 1 above)
Comment (ruby-oujo) [reply dist/api_data.py:953]: Thanks Sue! I have removed this now and will run the tests again.

**Comment by ruby-oujo** [comment]
Closing branch because of bad rebase, will open a new branch.

**Pull request closed without merging**